src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs

```diff
@@ -0,0 +1,270 @@
+			set { SetValue(MediaPlayerProperty, value); }
+		}
+
+		public static DependencyProperty MediaPlayerProperty { get; } =
+			DependencyProperty.Register(
+				nameof(MediaPlayer),
+				typeof(Windows.Media.Playback.MediaPlayer),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(default(Windows.Media.Playback.MediaPlayer), OnMediaPlayerChanged));
+
+		private static void OnMediaPlayerChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+		{
+			sender.Maybe<MediaPlayerElement>(mpe =>
+			{
+				if (args.OldValue is Windows.Media.Playback.MediaPlayer oldMediaPlayer)
+				{
+					oldMediaPlayer.MediaFailed -= mpe.OnMediaFailed;
+					oldMediaPlayer.MediaFailed -= mpe.OnMediaOpened;
+				}
+
+				if (args.NewValue is Windows.Media.Playback.MediaPlayer newMediaPlayer)
+				{
+					newMediaPlayer.Source = mpe.Source;
+					newMediaPlayer.MediaFailed += mpe.OnMediaFailed;
+					newMediaPlayer.MediaOpened += mpe.OnMediaOpened;
+					mpe.TransportControls?.SetMediaPlayer(newMediaPlayer);
+					mpe._isTransportControlsBound = true;
+				}
+			});
+		}
+
+		private void OnMediaFailed(Windows.Media.Playback.MediaPlayer session, object args)
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+			{
+				if (PosterSource != null && _posterImage != null)
+				{
+					_posterImage.Visibility = Visibility.Visible;
+					_mediaPlayerPresenter.Opacity = 0;
+				}
+			});
+		}
+
+		private void OnMediaOpened(Windows.Media.Playback.MediaPlayer session, object args)
+		{
+			Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+			{
+				if (_posterImage != null)
+				{
+					_posterImage.Visibility = Visibility.Collapsed;
+					_mediaPlayerPresenter.Opacity = 1;
+				}
+			});
+		}
+
+		#endregion
+
+		#region AreTransportControlsEnabled Property
+
+		public bool AreTransportControlsEnabled
+		{
+			get { return (bool)GetValue(AreTransportControlsEnabledProperty); }
+			set { SetValue(AreTransportControlsEnabledProperty, value); }
+		}
+
+		public static DependencyProperty AreTransportControlsEnabledProperty { get; } =
+			DependencyProperty.Register(
+				nameof(AreTransportControlsEnabled),
+				typeof(bool),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(false));
+
+		#endregion
+
+		#region Stretch Property
+
+		public Stretch Stretch
+		{
+			get { return (Stretch)GetValue(StretchProperty); }
+			set { SetValue(StretchProperty, value); }
+		}
+
+		public static DependencyProperty StretchProperty { get; } =
+			DependencyProperty.Register(
+				nameof(Stretch),
+				typeof(Stretch),
+				typeof(MediaPlayerElement),
+				new FrameworkPropertyMetadata(Stretch.Uniform));
+
+		#endregion
+
+		public MediaTransportControls TransportControls { get; set; } = new MediaTransportControls();
+
+		public MediaPlayerElement() : base()
+		{
+		}
+
+		protected override void OnLoaded()
+		{
+			base.OnLoaded();
+
+			MediaPlayer.AutoPlay = AutoPlay;
```

Comment:
Can MediaPlayer be null?

Reply:
Not at this point

Reply:
Are you assuming OnApplyTemplate has been called? Be careful, OnLoaded will be called before OnApplyTemplate if Visibility == Collapsed.

Reply:
Null check added